### PR TITLE
Export CLANG_PATH so bindgen works with cross compilation

### LINKED ISSUE #164: export-esp.sh does not set CLANG_PATH which can confuse bindgen (and any crate using clang-sys)
## Bug description

export-esp.sh is missing the CLANG_PATH variable which is used by clang-sys to locate the clang runtime.  bindgen uses this to determine -isystem.  Without it, the host's include path will be used while the target's clang toolchain is used.

- Would you like to work on a fix? Yes

## To Reproduce

```
. ~/export-esp.sh
export TARGET_CC=riscv32-esp-elf-gcc
export TARGET_CXX=riscv32-esp-elf-g++
export TARGET_AR=riscv32-esp-elf-ar
cd <some-esp32-project>
cargo add lvgl --git https://github.com/jasta/lvgl-rs --branch esp32-fixup
cargo add lvgl-sys --git https://github.com/jasta/lvgl-rs --branch esp32-fixup
cargo add lvlg-codegen --git https://github.com/jasta/lvgl-rs --branch esp32-fixup
# Download https://github.com/rafaelcaricio/lvgl-rs/blob/master/lvgl-sys/vendor/include/lv_conf.h and store in CWD.
DEP_LV_CONFIG_PATH=$(pwd) cargo +nightly build --target riscv32imc-esp-espidf
```

Lots of build errors related to stdint.h types missing (like int16_t).  Similar problems are likely to occur with other crates utilizing bindgen, but not always consistently.  It may also repro differently depending on the host and target clang mismatches for the user testing it.

## Expected behavior

Should compile :)

## Screenshots

N/A

## Environment

- OS: Ubuntu 22.04
- espup version: 0.2.5

## Additional context

Part of a larger body of work to fix cc and bindgen (a couple other issues block general purpose use cases)

### PR DESCRIPTION
**Pull request by jasta**
Fixes #164

CLANG_PATH is relevant for bindgen to be able to find required headers and library files.  Without exporting this along with LIBCLANG_PATH, you'll get a mismatch using bindgen and if the host clang doesn't support full cross compilation of C code (which is common for Ubuntu in my case, at least), then ultimately bindgen will not be able to compile the code its trying to generate bindings for.  The user will see weird errors like basic types int16_t and such are missing or common header files like string.h aren't available.

### CODE REVIEW

**Comment by jasta** [comment]
BTW I ran into an issue testing this PR on the latest main branch: 

https://gist.github.com/2fbeb823580cff6b56994d68f0720904

File: src/toolchain/llvm.rs [hunk 1]
```diff
@@ -53,6 +55,14 @@ impl Llvm {
         llvm_path
     }
 
+    fn get_bin_path(&self) -> String {
```
Comment (SergioGasquez) [inline src/toolchain/llvm.rs:58]:
```suggestion
    /// Gets the binary path of clang
    fn get_bin_path(&self) -> String {
```

**Review by SergioGasquez — COMMENTED** [review]
Thanks for your contribution! I just left a few minor comments, and, rebasing should fix the CI issues.

File: src/toolchain/llvm.rs (hunk 1 above)
Comment (jasta) [reply src/toolchain/llvm.rs:58]: I'm not sure I quite grok the meaning of using bin on Windows in the first place, I'll prefer to just keep it as-is unless you feel strongly

**Pull request merged**